ttnn/cpp/ttnn/operations/data_movement/concat/concat.cpp

```diff
@@ -97,36 +110,60 @@ MassagedConcat build_untilize_rm_retilize_concat(uint8_t queue_id, const MemoryC
             concat_db_print(res, "untilize_rm_retilize required");
             return res;
         },
-        .pre_transform = [](const std::vector<ttnn::Tensor>& tensors, int dim, unsigned int groups) -> OwnedConcatArgs {
+        .pre_transform =
+            [=](const std::vector<ttnn::Tensor>& tensors, int dim, unsigned int groups) -> OwnedConcatArgs {
```

Comment:
can we avoid capturing everything by copy? 
can we reference what we need by copy or by ref?

Reply:
maybe if we capture logical_output_shape by ref, we don't need shared_ptr

Reply:
See my other comment on the shared_ptr. I can make the capture list explicit if necessary.